### PR TITLE
Introduce FakeResourceLoader

### PR DESCRIPTION
**Pull request by swankjesse**
Always inject ResourceLoader, and use either the ResourceLoaderModule
or the FakeResourceLoaderModule.

This is so we don't need a bunch of resource files to test schema
migrations.

### CODE REVIEW

File: misk/src/test/kotlin/misk/jdbc/DataSourceModuleTest.kt [hunk 1]
```diff
@@ -153,7 +154,7 @@ internal class DataSourceModuleTest {
         |""".trimMargin()
 
       val DROP_TABLE_PEOPLE = """
-        |DROP TABLE people
+        |DROP TABLE IF EXISTS people
```
Comment (tirsen) [inline misk/src/test/kotlin/misk/jdbc/DataSourceModuleTest.kt]: Ohhhh multi line strings. I LOVE YOU KOTLIN!!!

**Comment by tirsen** [comment]
LGTM

File: misk/src/main/kotlin/misk/resources/ClasspathResourceLoader.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.TreeMap
+
+/** Real implementation of [ResourceLoader] suitable for development and production. */
+internal object ClasspathResourceLoader : ResourceLoader() {
+  private val resourcesByPath: Map<String, ClassPath.ResourceInfo>
+
+  init {
+    val classLoader = ResourceLoader::class.java.classLoader
+    val classPath = ClassPath.from(classLoader)
+    resourcesByPath = TreeMap(classPath.resources.associateBy { r -> r.resourceName })
```
Comment (shellderp) [inline misk/src/main/kotlin/misk/resources/ClasspathResourceLoader.kt]: use `it`?

File: misk/src/main/kotlin/misk/resources/ResourceLoaderModule.kt [hunk 3]
```diff
@@ -4,6 +4,6 @@ import misk.inject.KAbstractModule
 
 class ResourceLoaderModule : KAbstractModule() {
   override fun configure() {
-    bind<ResourceLoader>().toInstance(ResourceLoader)
+    bind<ResourceLoader>().toInstance(ClasspathResourceLoader)
```
Comment (shellderp) [inline misk/src/main/kotlin/misk/resources/ResourceLoaderModule.kt]: this doesn't need `()`? :O

File: misk/src/main/kotlin/misk/resources/ClasspathResourceLoader.kt (hunk 2 above)
Comment (swankjesse) [reply misk/src/main/kotlin/misk/resources/ClasspathResourceLoader.kt]: Fix in follow-up!

**Pull request merged**